packages/mesos-client/src/observableFromXhrConnection.js

```diff
@@ -0,0 +1,30 @@
+import { ConnectionConstants, ConnectionManager } from "connection-manager";
+import Rx from "rxjs";
+
+export default function observableFromXhrConnection(connection) {
+  return Rx.Observable.create(function(observer) {
+    connection.addListener(
+      ConnectionConstants.CONNECTION_EVENT_PROGRESS,
+      function(target) {
+        observer.next(target.response);
+      }
+    );
+    connection.addListener(ConnectionConstants.CONNECTION_EVENT_ERROR, function(
+      error
+    ) {
+      observer.error(error);
+    });
+    connection.addListener(
+      ConnectionConstants.CONNECTION_EVENT_CLOSE,
+      function() {
+        observer.complete();
+      }
+    );
+
+    ConnectionManager.queue(connection);
```

Comment:
I don't think this should live here - it pollutes the scope. It feels like magic because I would still expect to do this after I've defined my `XHRConnection` in `createEventStreamConnection.js`

Reply:
It may feel like a magic when you use it directly, but that's gonna be a utility function and the interface in general is that you pass the `baseUrl` and you have everything done for you. 

My first take was that one creates a connection and passes it to the observable wrapper that library provides. But then why would you call the library `mesos-client`, when you have to manually setup a connection? So after an offline discussion with @orlandohohmeier we've decided that it make much more sense to request only `baseUrl` and let the library do the rest.

Reply:
We probably have to move this helper to a different package and find a better name to make sure it feels less magic. How about introducing a resource(?) service that provides method similar to `fetch`  or`reqwest`which will internally create and enqueue a connection and return an observable.

This method needs to either clone the connection or create the connection (which is preferable) within the `Observable.create` closure as it otherwise probably doesn't make any sense to enqueue the connection. Only when we completely handing the connection management within the closure we can enable use cases like the following one:

```js
ResourceService
    .request("url", {method: "GET"})
    .retry(3)
    .subscribe((event) => console.log(event));
```